src/test/java/core/basesyntax/service/ValidatorTest.java

```diff
@@ -0,0 +1,61 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.db.StorageImpl;
+import core.basesyntax.dto.Transaction;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static core.basesyntax.model.OperationEnum.BALANCE;
+import static core.basesyntax.model.OperationEnum.PURCHASE;
+import static org.junit.jupiter.api.Assertions.*;
+
+class ValidatorTest {
+
+    private static final Storage storage = new StorageImpl();
+
+    @BeforeEach
+    public void init(){
+        storage.clear();
+    }
+
+    @Test
+    void validation_wrongQuantityProvided_failure() {
+        storage.saveToStorage("banana", 20);
+
+        assertThrows(WrongQuantityException.class, () ->
+                Validator.validate(new Transaction(PURCHASE,
+                        "banana",
+                        21),
+                        storage));
+    }
+
+    @Test
+    void validation_emptyStorageProvided_failure() {
+
+        assertThrows(WrongQuantityException.class, () ->
+                Validator.validate(new Transaction(PURCHASE,
+                                "banana",
+                                20),
+                        storage));
+    }
+
+    @Test
+    void validation_zeroQuantityProvided_failure() {
+
+        assertThrows(WrongQuantityException.class, () ->
+                Validator.validate(new Transaction(BALANCE,
+                                "banana",
+                                0),
+                        storage));
+    }
+
+    @Test
+    void validation_rightQuantityProvided_failure() throws WrongQuantityException {
+        storage.saveToStorage("banana", 100);
+        storage.saveToStorage("apple", 30);
+
+        Validator.validate(new Transaction(PURCHASE, "banana", 50), storage);
+        Validator.validate(new Transaction(PURCHASE, "apple", 10), storage);
+    }
```

Comment:
1. You can use ```assertDoesNotThrow``` method to verify that if right quantity provided - no exception is thrown
2. Please update the method name, for this case it should be ```_success``` probably :blush:

Reply:
Thanks for your suggestion, should I in all tests names add _success if the right data provided, or I can just remove _failure from this method?

Reply:
Let's add ```_success``` for positive test-cases